pkg/middlewares/aws_route_table_expander_test.go

```diff
@@ -208,3 +211,85 @@ func TestAwsRouteTableExpander_Execute(t *testing.T) {
 		})
 	}
 }
+
+func TestAwsRouteTableExpander_ExecuteWithInvalidRoutes(t *testing.T) {
+
+	mockedAlerter := &mocks.AlerterInterface{}
+	mockedAlerter.On("SendAlert", aws.AwsRouteTableResourceType, alerter.Alert{
+		Message: "Skipped invalid route found in state for aws_route_table.table_from_state",
+	})
+	mockedAlerter.On("SendAlert", aws.AwsDefaultRouteTableResourceType, alerter.Alert{
+		Message: "Skipped invalid route found in state for aws_default_route_table.default_table_from_state",
+	})
```

Comment:
Why do you mock here the alerter instead of calling `alerter := alerter.NewAlerter()` and assert that at the end we have indeed the two alerts in its map of alerts ? 

![image](https://user-images.githubusercontent.com/8110579/107017954-20f09c80-67a0-11eb-8e5a-d52b4a5e84ec.png)

Reply:
The scope of this test is not to test the alerter itself but only the behaviour of the middleware handling this case.

Unit test should stay minimalist as possible IMHO

Reply:
Make sense